src/w3d/renderer/assetmgr.cpp

```diff
@@ -746,13 +749,62 @@ RenderObjClass *GameAssetManager::Create_Render_Obj(
     return robj;
 }
 
+bool Get_Mesh_Color_Methods(MeshClass *mesh, bool *housecolor, bool *zhc)
+{
+    *housecolor = false;
+    *zhc = false;
+    MaterialInfoClass *matinfo = mesh->Get_Material_Info();
+
+    if (matinfo) {
+        for (int i = 0; i < matinfo->Texture_Count(); i++) {
+
+            if (!strncasecmp(matinfo->Peek_Texture(i)->Get_Name(), "ZHC", 3)) {
+                *zhc = true;
+                break;
+            }
+        }
+
+        matinfo->Release_Ref();
+    }
+
+    const char *str = strchr(mesh->Get_Name(), '.');
+    const char *str2;
+
+    if (str && (str2 = str + 1, *str) || (str2 = mesh->Get_Name()) != 0) {
+        if (!strncasecmp(str2, "HOUSECOLOR", 10)) {
+            *housecolor = true;
+        }
+    }
+
+    return *housecolor || *zhc;
+}
+
 // 0x00765DF0
 void GameAssetManager::Make_Mesh_Unique(RenderObjClass *robj, bool geometry, bool colors)
 {
-    // Uses MeshClass requires MaterialInfoClass
-#ifdef GAME_DLL
-    Call_Method<void, W3DAssetManager, RenderObjClass *, bool, bool>(0x00765DF0, this, robj, geometry, colors);
-#endif
+    MeshClass *mesh = static_cast<MeshClass *>(robj);
+    bool housecolor;
+    bool zhc;
+
+    if (colors && Get_Mesh_Color_Methods(mesh, &housecolor, &zhc) || geometry) {

```

Comment:
Could you add extra () around this to make it completely unambiguous to the reader.

Reply:
fixed.